engine/table/src/main/java/io/deephaven/engine/table/impl/select/WhereFilter.java

```diff
@@ -211,161 +196,20 @@ public PreviousFilteringNotSupported(String message) {
-                return preferred.lhs().walk(new FilterConditionAdapter(condition, preferred)).getOut();
-            }
-
-            private final FilterCondition original;
-            private final FilterCondition preferred;
-
-            private WhereFilter out;
-
-            private FilterConditionAdapter(FilterCondition original, FilterCondition preferred) {
-                this.original = Objects.requireNonNull(original);
-                this.preferred = Objects.requireNonNull(preferred);
-            }
-
-            public WhereFilter getOut() {
-                return Objects.requireNonNull(out);
-            }
-
-            @Override
-            public void visit(ColumnName lhs) {
-                preferred.rhs().walk(new Value.Visitor() {
-                    @Override
-                    public void visit(ColumnName rhs) {
-                        out = WhereFilterFactory.getExpression(Strings.of(original));
-                    }
-
-                    @Override
-                    public void visit(long rhs) {
-                        switch (preferred.operator()) {
-                            case LESS_THAN:
-                                out = new LongRangeFilter(lhs.name(), Long.MIN_VALUE, rhs, true, false);
-                                break;
-                            case LESS_THAN_OR_EQUAL:
-                                out = new LongRangeFilter(lhs.name(), Long.MIN_VALUE, rhs, true, true);
-                                break;
-                            case GREATER_THAN:
-                                out = new LongRangeFilter(lhs.name(), rhs, Long.MAX_VALUE, false, true);
-                                break;
-                            case GREATER_THAN_OR_EQUAL:
-                                out = new LongRangeFilter(lhs.name(), rhs, Long.MAX_VALUE, true, true);
-                                break;
-                            case EQUALS:
-                                out = new MatchFilter(lhs.name(), rhs);
-                                break;
-                            case NOT_EQUALS:
-                                out = new MatchFilter(MatchType.Inverted, lhs.name(), rhs);
-                                break;
-                            default:
-                                throw new IllegalStateException("Unexpected operator " + original.operator());
-                        }
-                    }
-                });
-            }
-
-            // Note for all remaining cases: since we are walking the preferred object, we know we don't have to handle
-            // the case where rhs is column name.
+    // region Filter impl
 
-            @Override
-            public void visit(long lhs) {
-                out = WhereFilterFactory.getExpression(Strings.of(original));
-            }
-        }
+    @Override
+    default Filter invert() {
+        throw new UnsupportedOperationException("WhereFilters do not implement invert");
```

Comment:
just a curiosity / skepticism issue, but is this thing really an Expression or Filter if it can't implement `invert`, `walk(Expression.Visitor)`, or `walk(Filter.Visitor)`

Reply:
It's kind of a crappy hack; `WhereFilter` is not *really* a `io.deephaven.api.filter.Filter`, but decisions were made so that they would work in the context of `io.deephaven.engine.table.Table#where(Filter... filters)`.